ztest/ztest.go

```diff
@@ -121,66 +169,114 @@ func Run(t *testing.T, dirname string) {
 			if err != nil {
 				t.Fatalf("%s: %s", filename, err)
 			}
-			out, errout, err := run(zq, zt.ZQL, zt.OutputFormat, zt.OutputFlags, zt.Input...)
-			if err != nil {
-				if zt.errRegex != nil {
-					if !zt.errRegex.Match([]byte(errout)) {
-						t.Fatalf("%s: error doesn't match expected error regex: %s %s", filename, zt.ErrorRE, errout)
-					}
-				} else {
-					if out != "" {
-						out = "\noutput:\n" + out
-					}
-					t.Fatalf("%s: %s%s", filename, err, out)
-				}
-			} else if zt.errRegex != nil {
-				t.Fatalf("%s: no error when expecting error regex: %s", filename, zt.ErrorRE)
-			}
-			expectedOut, oerr := zt.getOutput()
-			require.NoError(t, oerr)
-			if out != expectedOut {
-				a := expectedOut
-				b := out
+			run(t, testname, bindir, dirname, filename, zq, zt)
+		})
+	}
+}
 
-				if !utf8.ValidString(a) {
-					a = encodeHex(a)
-					b = encodeHex(b)
-				}
+type File struct {
+	// Name is the name of the file in which the test shell code runs.
```

Comment:
```suggestion
	// Name is the base file name for this data.
```

Reply:
I added more clarification because it's a bit overloaded.